config/log_test.go

```diff
@@ -0,0 +1,343 @@
+					MaxQueueSize:       ptr(0),
+					ScheduleDelay:      ptr(0),
+					Exporter: LogRecordExporter{
+						OTLP: &OTLP{
+							Protocol:    "http/protobuf",
+							Compression: ptr("gzip"),
+							Timeout:     ptr(1000),
+							Headers: map[string]string{
+								"test": "test1",
+							},
+						},
+					},
+				},
+			},
+			wantProcessor: sdklog.NewBatchProcessor(otlpHTTPExporter),
+		},
+		{
+			name: "batch/otlp-http-exporter-no-scheme",
+			processor: LogRecordProcessor{
+				Batch: &BatchLogRecordProcessor{
+					MaxExportBatchSize: ptr(0),
+					ExportTimeout:      ptr(0),
+					MaxQueueSize:       ptr(0),
+					ScheduleDelay:      ptr(0),
+					Exporter: LogRecordExporter{
+						OTLP: &OTLP{
+							Protocol:    "http/protobuf",
+							Endpoint:    "localhost:4318",
+							Compression: ptr("gzip"),
+							Timeout:     ptr(1000),
+							Headers: map[string]string{
+								"test": "test1",
+							},
+						},
+					},
+				},
+			},
+			wantProcessor: sdklog.NewBatchProcessor(otlpHTTPExporter),
+		},
+		{
+			name: "batch/otlp-http-invalid-endpoint",
+			processor: LogRecordProcessor{
+				Batch: &BatchLogRecordProcessor{
+					MaxExportBatchSize: ptr(0),
+					ExportTimeout:      ptr(0),
+					MaxQueueSize:       ptr(0),
+					ScheduleDelay:      ptr(0),
+					Exporter: LogRecordExporter{
+						OTLP: &OTLP{
+							Protocol:    "http/protobuf",
+							Endpoint:    " ",
+							Compression: ptr("gzip"),
+							Timeout:     ptr(1000),
+							Headers: map[string]string{
+								"test": "test1",
+							},
+						},
+					},
+				},
+			},
+			wantErr: &url.Error{Op: "parse", URL: " ", Err: errors.New("invalid URI for request")},
+		},
+		{
+			name: "batch/otlp-http-none-compression",
+			processor: LogRecordProcessor{
+				Batch: &BatchLogRecordProcessor{
+					MaxExportBatchSize: ptr(0),
+					ExportTimeout:      ptr(0),
+					MaxQueueSize:       ptr(0),
+					ScheduleDelay:      ptr(0),
+					Exporter: LogRecordExporter{
+						OTLP: &OTLP{
+							Protocol:    "http/protobuf",
+							Endpoint:    "localhost:4318",
+							Compression: ptr("none"),
+							Timeout:     ptr(1000),
+							Headers: map[string]string{
+								"test": "test1",
+							},
+						},
+					},
+				},
+			},
+			wantProcessor: sdklog.NewBatchProcessor(otlpHTTPExporter),
+		},
+		{
+			name: "batch/otlp-http-invalid-compression",
+			processor: LogRecordProcessor{
+				Batch: &BatchLogRecordProcessor{
+					MaxExportBatchSize: ptr(0),
+					ExportTimeout:      ptr(0),
+					MaxQueueSize:       ptr(0),
+					ScheduleDelay:      ptr(0),
+					Exporter: LogRecordExporter{
+						OTLP: &OTLP{
+							Protocol:    "http/protobuf",
+							Endpoint:    "localhost:4318",
+							Compression: ptr("invalid"),
+							Timeout:     ptr(1000),
+							Headers: map[string]string{
+								"test": "test1",
+							},
+						},
+					},
+				},
+			},
+			wantErr: errors.New("unsupported compression \"invalid\""),
+		},
+		{
+			name: "simple/no-exporter",
```

Comment:
Missing test for simple with a valid exporter.

Reply:
Added